src/pylint/rules/magic_value_comparison.rs

```diff
@@ -0,0 +1,41 @@
+use rustpython_ast::{Constant, Expr, ExprKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violations;
+
+fn is_magic_value(constant: &Constant) -> Option<String> {
+    match constant {
+        Constant::Int(value) => {
+            if matches!(value.try_into(), Ok(-1 | 0 | 1)) {
+                None
+            } else {
+                Some(value.to_string())
+            }
+        }
+        Constant::Float(value) => Some(value.to_string()),
+        _ => None,
+    }
+}
+
+/// PLR2004
+pub fn magic_value_comparison(
+    checker: &mut Checker,
+    expr: &Expr,
+    left: &Expr,
+    comparators: &[Expr],
+) {
+    for comparison_expr in comparators.iter().chain([left]) {
+        if let ExprKind::Constant { value, .. } = &comparison_expr.node {
```

Comment:
I think we may need to avoid comparisons between _two_ constants?

https://github.com/PyCQA/pylint/blob/0c5ab132e262280da98742e5cf73ecc1cbfa7377/pylint/extensions/magic_value.py#L81

Reply:
Good catch, have added a comment to hopefully better explain why the rule is skipped.